packages/common/tests/mergePOS.spec.ts

```diff
@@ -187,4 +189,122 @@ tape('[Common]: Merge/POS specific logic', function (t: tape.Test) {
     st.equal(c.getHardforkByBlockNumber(5, 0), 'shanghai', msg)
     st.end()
   })
+
+  t.test('should get the correct merge hardfork at genesis', async (st) => {
+    const json = require(`../../client/test/testdata/geth-genesis/post-merge.json`)
+    const c = Common.fromGethGenesis(json, { chain: 'post-merge' })
+    const msg = 'should get HF correctly'
+    st.equal(c.getHardforkByBlockNumber(0), Hardfork.London, msg)
+    st.equal(c.getHardforkByBlockNumber(0, BigInt(0)), Hardfork.Merge, msg)
+  })
+
+  t.test('test post merge hardforks using Sepolia with block null', function (st: tape.Test) {
+    const c = new Common({ chain: Chain.Sepolia })
+    let msg = 'should get HF correctly'
+
+    st.equal(c.getHardforkByBlockNumber(0), Hardfork.London, msg)
+    // Make it null manually as config could be updated later
+    const mergeHf = c.hardforks().filter((hf) => hf.ttd !== undefined && hf.ttd !== null)[0]
+    const prevMergeBlockVal = mergeHf.block
+    mergeHf.block = null
+
+    // should get Hardfork.London even though happened with 1450408 as terminal as config doesn't have that info
```

Comment:
@acolytec3  see if these comments help making sense of the situation

Reply:
yes but for this test case i made block null, see L209, i have made this test to be independent of what we specify config in json